internal/generate/proto/generate.go

```diff
@@ -5,45 +5,36 @@ import (
 	"fmt"
 	"log"
 	"os/exec"
+	"path/filepath"
 	"sync"
 
 	"github.com/commitdev/commit0/internal/config"
 	"github.com/commitdev/commit0/internal/templator"
 	"github.com/commitdev/commit0/internal/util"
 )
 
-func Generate(templator *templator.Templator, config *config.Commit0Config, wg *sync.WaitGroup) {
-	idlPath := fmt.Sprintf("%s-idl", config.Name)
-	idlHealthPath := fmt.Sprintf("%s/proto/health", idlPath)
+func Generate(t *templator.Templator, cfg *config.Commit0Config, service config.Service, wg *sync.WaitGroup) {
+	idlPath := fmt.Sprintf("%s-idl", service.Name)
+	idlHealthPath := filepath.Join(idlPath, "proto", "health")
 
-	util.TemplateFileIfDoesNotExist(idlPath, "go.mod", templator.Go.GoModIDL, wg, config)
-	util.TemplateFileIfDoesNotExist(idlPath, "Makefile", templator.MakefileTemplate, wg, config)
-	GenerateServiceProtobufFiles(templator, config, wg)
-	util.TemplateFileIfDoesNotExist(idlHealthPath, "health.proto", templator.ProtoHealthTemplate, wg, config)
+	data := templator.GolangTemplateData{
+		*cfg,
+		service,
+	}
 
-	GenerateProtoServiceLibs(config)
-}
+	util.TemplateFileIfDoesNotExist(idlPath, "go.mod", t.Go.GoModIDL, wg, data)
+	util.TemplateFileIfDoesNotExist(idlPath, "Makefile", t.MakefileTemplate, wg, data)
+	util.TemplateFileIfDoesNotExist(idlHealthPath, "health.proto", t.ProtoHealthTemplate, wg, data)
 
-func GenerateServiceProtobufFiles(templator *templator.Templator, cfg *config.Commit0Config, wg *sync.WaitGroup) {
-	protoPath := fmt.Sprintf("%s-idl/proto", cfg.Name)
-	for _, s := range cfg.Services {
-		serviceProtoDir := fmt.Sprintf("%s/%s", protoPath, s.Name)
-		file := fmt.Sprintf("%s.proto", s.Name)
-
-		data := struct {
-			*config.Commit0Config
-			ServiceName string
-		}{
-			cfg,
-			s.Name,
-		}
-
-		util.TemplateFileIfDoesNotExist(serviceProtoDir, file, templator.ProtoServiceTemplate, wg, data)
-	}
+	serviceProtoDir := filepath.Join(idlPath, "proto", service.Name)
+	file := fmt.Sprintf("%s.proto", service.Name)
+	util.TemplateFileIfDoesNotExist(serviceProtoDir, file, t.ProtoServiceTemplate, wg, data)
+
+	GenerateProtoServiceLibs(service)
 }
 
-func GenerateProtoServiceLibs(config *config.Commit0Config) {
-	idlRoot := fmt.Sprintf("%s-idl", config.Name)
+func GenerateProtoServiceLibs(service config.Service) {
```

Comment:
Before the idl repo produced would be a single repo that contained all the service idls & generated code split by folders, mainly for sharing protos (such as health) and for ease of use - is the intention to change this to multiple repos?